lib/sequelize.js

```diff
@@ -342,8 +342,12 @@ class Sequelize {
     /**
      * Models are stored here under the name given to `sequelize.define`
      */
-    this.models = {};
-    this.modelManager = new ModelManager(this);
+    this.modelManager = new ModelManager();
+    this.models = new Proxy(this.modelManager, {
```

Comment:
While I was the first one to introduce proxies to sequelize, I'd rather we don't do this.
Proxies are _slow_ and shouldn't really be used at runtime (startup time is usually fine).